tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -948,24 +799,114 @@ def test_split_order(direction: str, constant_liar: bool, constraints: bool) ->
     else:
         states = [optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.PRUNED]
 
-    scores, violations = _tpe.sampler._get_observation_pairs(
-        study,
-        study.get_trials(states=states),
-        constraints,
+    trials = study.get_trials(states=states)
+    finished_trials = study.get_trials(
+        states=(optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.PRUNED)
     )
+    for n_below in range(len(finished_trials) + 1):
+        below_trials, above_trials = _tpe.sampler._split_trials(
+            study,
+            trials,
+            n_below,
+            constraints,
+        )
+
+        below_trial_numbers = [trial.number for trial in below_trials]
+        assert below_trial_numbers == list(range(n_below))
+        above_trial_numbers = [trial.number for trial in above_trials]
+        assert above_trial_numbers == list(range(n_below, len(trials)))
 
-    assert len(scores) == len(study.get_trials(states=states))
-    if constraints:
-        assert violations is not None
-        assert len(violations) == len(study.get_trials(states=states))
-    else:
-        assert violations is None
 
-    for gamma in range(1, len(scores)):
-        indices_below, indices_above = _tpe.sampler._split_observation_pairs(
-            scores, gamma, violations
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+def test_split_complete_trials_single_objective(direction: str) -> None:
+    study = optuna.create_study(direction=direction)
+
+    for value in [-float("inf"), 0, 1, float("inf")]:
+        study.add_trial(
+            optuna.create_trial(
+                state=optuna.trial.TrialState.COMPLETE,
+                value=(value if direction == "minimize" else -value),
+                params={"x": 0},
+                distributions={"x": optuna.distributions.FloatDistribution(-1.0, 1.0)},
+            )
         )
-        assert np.max(indices_below) < np.min(indices_above)
+
+    for n_below in range(len(study.trials) + 1):
+        below_trials, above_trials = _tpe.sampler._split_complete_trials_single_objective(
+            study.trials,
+            study,
+            n_below,
+        )
+        assert [trial.number for trial in below_trials] == list(range(n_below))
+        assert [trial.number for trial in above_trials] == list(range(n_below, len(study.trials)))
+
+
+def test_split_complete_trials_single_objective_empty() -> None:
+    study = optuna.create_study()
+    _tpe.sampler._split_complete_trials_single_objective([], study, 0) == ([], [])
```

Comment:
```suggestion
    assert _tpe.sampler._split_complete_trials_single_objective([], study, 0) == ([], [])
```

Reply:
Thank you, good catch! I fixed the missing assertion. PTAL.